server/data/data-file/src/main/java/org/apache/james/rrt/file/XMLRecipientRewriteTable.java

```diff
@@ -69,16 +69,10 @@ protected Mappings mapAddress(String user, Domain domain) {
 
     @Override
     public Mappings getUserDomainMappings(MappingSource source) {
-        if (mappings == null) {
-            return null;
-        } else {
-            String maps = mappings.get(source);
-            if (maps != null) {
-                return MappingsImpl.fromRawString(maps);
-            } else {
-                return null;
-            }
-        }
+        return Optional.ofNullable(mappings)
+                .map(mappings -> mappings.get(source))
+                .map(MappingsImpl::fromRawString)
+                .orElse(MappingsImpl.empty());
```

Comment:
as you want `getUserDomainMappings` method always return a non null value. I guess you can refactor some other places where we use that method (perhaps in some Routes). 

And I agree with the ideal to make that method returns a non null value. How about marking that method with `@Nonnull` annotation and let the callers know the value returned will not be null ever?